pykern/pkcli/fmt.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+u"""Wrapper for Python formatter (currently, ``black``) to update and to validate a repository.
+
+:copyright: Copyright (c) 2015 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern.pkdebug import pkdp
+import pykern.pksubprocess
+import pykern.pkunit
+
+def run(path):
+    '''
+        Run black formatter on path
+
+        Args:
+            path - string path to file or dir
+    '''
+    pykern.pksubprocess.check_call_with_signals(['black', f'{path}'])
+
+
+def diff(path):
+    pykern.pksubprocess.check_call_with_signals([
+        'black',
+        '--diff',
+        '--no-color',
+        f'{path}'
+    ])
+
+
+
+def check(path):
+    '''Returns True if there would be diff else return False'''
+    try:
+        pykern.pksubprocess.check_call_with_signals([
+            'black',
+            '--check',
+            f'{path}'
+        ])
+    except RuntimeError as e:
+        if str(e) == 'error exit(1)':
```

Comment:
I don't like the str, but it's ok given the interface. Returning True is not how you return an error (check fails). You have to call command_error. Also, call `raise` when you don't get what you expect (another exception):

```py
except RuntimeError as e:
        if  str(e) == "error exit(1)":
            pykern.pkcli.command_error('path={} needs to be formatted', path)
        raise
```

Reply:
This function just checks if their is diff. So do we really want to raise an error every time there is diff ('error exit(1)' being the one `black check ...` raises) or do we just want to use check to return True or False that their will be diff?